microservice/security/src/main/java/com/cumulocity/microservice/security/filter/provider/SpringSecurityContextProvider.java

```diff
@@ -5,28 +5,22 @@
 import com.cumulocity.microservice.context.credentials.UserCredentials;
 import com.cumulocity.microservice.security.filter.util.HttpRequestUtils;
 import com.cumulocity.microservice.subscription.service.MicroserviceSubscriptionsService;
-import java.util.Optional;
+import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.beans.factory.annotation.Value;
 import org.springframework.security.access.AccessDeniedException;
 import org.springframework.security.core.context.SecurityContext;
 import org.springframework.security.core.userdetails.UserDetails;
-import org.springframework.stereotype.Component;
 import org.springframework.util.StringUtils;
 
+import java.util.Optional;
+
 @Slf4j
-@Component
+@RequiredArgsConstructor
```

Comment:
minor: maybe declare constructor directly (without Lombok), as you do in the other place (`CumulocityOAuthMicroserviceFilter`)